srcpkgs/ccache/template

```diff
@@ -1,19 +1,22 @@
 # Template file for 'ccache'
 pkgname=ccache
-version=3.7.12
+version=4.0
 revision=1
 bootstrap=yes
-build_style=gnu-configure
-make_check_args="CC=gcc"
-makedepends="zlib-devel"
+build_style=cmake
+makedepends="libzstd-devel zlib-devel"
 checkdepends="perl"
 short_desc="Fast C/C++ Compiler Cache"
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="GPL-3.0-or-later"
 homepage="https://ccache.samba.org/"
 changelog="https://ccache.dev/releasenotes.html"
 distfiles="https://github.com/ccache/ccache/releases/download/v${version}/${pkgname}-${version}.tar.xz"
-checksum=a02f4e8360dc6618bc494ca35b0ae21cea080f804a4898eab1ad3fcd108eb400
+checksum=ac1b82fe0a5e39905945c4d68fcb24bd0f32344869faf647a1b8d31e544dcb88
+
+if [ "$CHROOT_READY" ]; then
```

Comment:
I was thinking about `stage1` and `stage2`.
Either put inside `chroot_ready` or set it unconditionally. I don't really know. I haven't put much thought into that. :shrug:

Reply:
As a rule of thumb you usually do not depend on anything `chroot-...` because it is, well, in the chroot anyway.
That's why I thought that perhaps you meant
```
if [ "$CHROOT_READY" -a "$CROSS_BUILD" ]; then
        hostmakedepends+=" cmake"
fi
```
We need to test that anyway... I'll close my PR, your's is better.

Reply:
This `chroot-cmake` is very basic `cmake`, it lacks a lot of `cmake` functionality (but enough to build `ccache` and real `cmake`).
I think we should treat it the same with `pkgconf` and `pkg-config`. IOW, don't add it into `base-chroot` but let's package add it into `hostmakedepends`.

Reply:
Ok, I see, then this makes sense.
Let's hope that ccache doesn't become too fancy with its cmake scripts in the next updates :)

Reply:
If we don't add it to `base-chroot` then a better name then would be something like `cmake-minimal` or `cmake-bootstrap`.

Reply:
OK, I'll change it into `cmake-bootstrap`

Reply:
Looking around, I think if `ccache` don't use those fancy  `ExternalProject`, we should be fine.

The whole `curl`, `libarchive`, `rhash`, `expat` and `nghttp2` was added for fetching tarball from external resource, verify checksum and build them.